### DISCUSSION TITLE
目前PCL服务器是干嘛用的？

### DISCUSSION THREAD

**Discussion by Deep-Dark-Forest** [Meloong-Git/PCL#3363] [提问] [answered]
要是有好心人能解释一下https://t.bilibili.com/890096306690719752?spm_id_from=333.999.list.card_time.click 我会不胜受恩感激（

**Comment by Ad-closeNN** [comment]
额这个服务器是日志里面的服务器吗
```shell
[19:07:44.626] [Net] 获取网络结果：https://pcl2-server-1253424809.file.myqcloud.com/notice.cfg?sign=1708254465-4e55f722-0-959dd9342accb58eb2daa229a3c4d87e，超时 10000ms
```
~~话说这个服务器是干嘛用的~~

**Reply by hejiehao** [reply]
谁去逝逝轰炸这个服务器

**Reply by mc-124** [reply]
有403的，炸不了

**Reply by Ad-closeNN** [reply]
这个服务器没权限访问
~~话说github.io能这么设权限吗~~

**Reply by allMagicNB** [reply]
> 这个服务器没权限访问
> ~~话说github.io能这么设权限吗~~

要带参数才行

**Comment by allMagicNB** [comment]
1.服务器公告
2.爱发电机器人
3.身份验证
4.QQ机器人
5.内群身份验证
6.下载文件，比如特供下载，启动器更新

**Reply by Deep-Dark-Forest** [reply]
那能否解释一下https://t.bilibili.com/890096306690719752?spm_id_from=333.999.list.card_time.click
` [B 站不让发的那个网站] ` 又是什么

**Reply by allMagicNB** [reply]
> 那能否解释一下https://t.bilibili.com/890096306690719752?spm_id_from=333.999.list.card_time.click

爱发电的机器人和这里的服务器是同一个

**Reply by Ad-closeNN** [reply]
> > 那能否解释一下https://t.bilibili.com/890096306690719752?spm_id_from=333.999.list.card_time.click
> 
> 爱发电的机器人和这里的服务器是同一个

我还以为爱发电的机器人是爱发电提供的免费服务（跟B站的自动回复一样）

**Reply by hejiehao** [reply]
那这个 Bot 也太智能了

**Reply by allMagicNB** [reply]
> 那这个 Bot 也太智能了

应该是同一个服务器但是不同端口？Bot和服务器是分开的，但应该使用同一个数据库

**Reply by wuliaodexiaoluo** [reply]
猜测是 afdian.net

**Comment by Silverteal** [comment, marked as answer]
至少有2个“服务器”。
* 第一个是“腾讯云CVM”，维护了：
  * 爱发电回复机器人
  * 内群身份验证
  * 本仓库的 Github 机器人
* 第二个是“腾讯云COS+腾讯云CDN”，提供了：
  * 启动器公告
  * 启动器更新文件
  * PCL2特供下载版本的元数据

**Reply by allMagicNB** [reply]
问的是功能（

**Reply by Silverteal** [reply]
COS其实不算服务器，如果非要像这样抠字眼的话，那应该只写CVM部分……

**Reply by wuliaodexiaoluo** [reply]
Hex-DragonBot 也是 CVM 负责的吧

**Reply by Silverteal** [reply]
> Hex-DragonBot 也是 CVM 负责的吧

似乎确实。我感觉本来可以用Github Actions的

**Reply by wuliaodexiaoluo** [reply]
Action 会引发一些问题，没控好可用挖矿，不需要 Member 同意

**Reply by Silverteal** [reply]
这里有什么场景会被用来挖矿的么
Action好歹还是Github托管运行器，照这么说CVM不是应该更危险么，没太懂您的意思。

**Reply by wuliaodexiaoluo** [reply]
理论上来说这只是其中一个可能
CVM 有登录鉴权，Action 不需要 member 同意合并 pull request 就能执行
可能二是 Action 的团队版只允许 3000 分钟，而 bot 是实时在线的

**Reply by Deep-Dark-Forest** [reply]
> Action 不需要 member 同意合并 pull request 就能执行

我不认同，因为settings里能改 Action 的权限

> 可能二是 Action 的团队版只允许 3000 分钟，而 bot 是实时在线的

我不认同，因为
![image](https://github.com/Hex-Dragon/PCL2/assets/131975043/4c21ea2b-3cca-4feb-8f52-80a5f7be289f)

**Reply by wuliaodexiaoluo** [reply]
不管他，反正龙猫这么做肯定有他的理由

**Reply by Silverteal** [reply]
> 理论上来说这只是其中一个可能
CVM 有登录鉴权，Action 不需要 member 同意合并 pull request 就能执行
可能二是 Action 的团队版只允许 3000 分钟，而 bot 是实时在线的

我不认为这里提出的是实际问题。我没有看出来“没有同意合并就执行工作流”有任何安全性的影响——而且，据我所知目前机器人并不操作仓库代码，也没有用到PR触发器。“登录CVM”更是和机器人自动化这一目的没有任何关系。Actions的运行限制对公开仓库是几乎没有的……更有可能的解释是，有一些操作无法通过Actions完成 ~~，或者更简单的，不会写工作流~~。

> 不管他，反正龙猫这么做肯定有他的理由

请问阁下的判断从何而来呢……

**Reply by wuliaodexiaoluo** [reply]
反正跟我们没关系就是了

**Reply by wuliaodexiaoluo** [reply]
> > 理论上来说这只是其中一个可能
> > CVM 有登录鉴权，Action 不需要 member 同意合并 pull request 就能执行
> > 可能二是 Action 的团队版只允许 3000 分钟，而 bot 是实时在线的
> 
> 我不认为这里提出的是实际问题。我没有看出来“没有同意合并就执行工作流”有任何安全性的影响——而且，据我所知目前机器人并不操作仓库代码，也没有用到PR触发器。“登录CVM”更是和机器人自动化这一目的没有任何关系。Actions的运行限制对公开仓库是几乎没有的……更有可能的解释是，有一些操作无法通过Actions完成 ~，或者更简单的，不会写工作流~。
> 
> > 不管他，反正龙猫这么做肯定有他的理由
> 
> 请问阁下的判断从何而来呢……

不会写工作流不太可能，有模板套，自己改改就能用，我不认为龙猫没有这样的能力
唯一合理的解释可能是 Action 不能创建讨论贴（新功能投票），Label Action 完全可以改，或者龙猫懒得写

**Reply by Deep-Dark-Forest** [reply]
> 唯一合理的解释可能是 Action 不能创建讨论贴（新功能投票），Label Action 完全可以改，或者龙猫懒得写

你说的对

```
name: Issue to Discussion

on:
  issues:
    types: [labeled]

jobs:
  create_discussion:
    if: github.event.label.name == 'bug'
    runs-on: ubuntu-latest
    steps:
      - name: Create Discussion
        uses: actions/github-script@v5
        with:
          script: |
            const title = "原神真好玩";
            const body = "崩铁真好玩";
            const repo = context.repo.repo;
            const owner = context.repo.owner;
            const token = process.env.GITHUB_TOKEN;
            
            const query = `
              mutation {
                createDiscussion(input: {
                  repositoryId: ${{ github.event.repository.node_id }},
                  categoryId: "YOUR_DISCUSSION_CATEGORY_ID",
                  title: "${title}",
                  body: "${body}"
                }) {
                  discussion {
                    url
                  }
                }
              }
            `;
            
            const headers = {
              "Authorization": `token ${token}`
            };
            
            const createDiscussion = await github.graphql(query, headers);
            
            console.log("Discussion created: ", createDiscussion.createDiscussion.discussion.url);
```

**Reply by SqrtTan** [reply]
~你们这斗法呢~
